### ISSUE TITLE
BPKG does not enforce the submodule update strategy

### ISSUE THREAD

**Issue by dalfy** [build2/bpkg#1]
Whenever packaging third party software with build2, I tend to setup a submodule named upstream that point to the commit that's package for build2. It allows to keep track and also to update the package to a newer version in a controlled way. 

At the moment when fetching the repository for the packaged software using bpkg. All submodules are recursively cloned and checked out ignoring the submodule policy. This increase significantly all package fetch and synchronisation of the dependencies of one project.

**Comment by dalfy** [comment]
Extracted discussion from Slack: 

I will try to produce the bare minimum repository that reproduce the problem. For the time being find below an example of the output of my CI system:
```querying https://github.com/dalfy/build2-third_party.git
fetching from https://github.com/dalfy/build2-third_party.git
synchronizing:
  new libglad/4.6.3 (required by dungeon)
  new libglm/0.9.9-a.4.2 (required by dungeon)
  new libcatch2/2.6.0 (required by dungeon)
  new libnuklear/0.1.0-a.1.20190311085243.104dc6436f07 (required by dungeon)
  new dungeon/0.1.0-a.0.20190316105904.8570a659b711
checking out libglad/4.6.3
querying https://github.com/bulletphysics/bullet3.git
fetching submodule 'libbullet3/upstream' from https://github.com/bulletphysics/bullet3.git
submodule path 'libbullet3/upstream': checked out '126b6762200b1a9ec8df39b8fd53cb2ad62fb146'
```

In the above the line were it goes "wrong" is fetching submodule, as in the submodule folder I have set a configuration entry of update = none on all the upstream project as those are only needed when one want to actually bring a new version. I have created the following repository that should reproduce the problem. https://github.com/dalfy/build2-git-submodule-issue

**Comment by karen-arutyunov** [comment]
Now bpkg skips git repository submodules with 'none' update method configured. 

This functionality will be available in 0.11.0 release. Until then you can use the staged version (https://stage.build2.org/0/0.11.0-a.0/).

Please reopen if still see any issues.
